metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -200,23 +201,24 @@ Builder setCreateTopicPolicy(Optional<CreateTopicPolicy> createTopicPolicy) {
         }
 
         ReplicationControlManager build() {
-            if (configurationControl == null) {
-                throw new RuntimeException("You must specify configurationControl.");
-            }
-            if (clusterControl == null) {
-                throw new RuntimeException("You must specify clusterControl.");
-            }
-            if (controllerMetrics == null) {
-                throw new RuntimeException("You must specify controllerMetrics.");
+            if (metadataVersion == null) {
+                throw new IllegalStateException("Metadata version must be set before building");
```

Comment:
Let's just default to the latest version. It's very tedious to have to set this in every single unit test (most of which don't care)

Reply:
Sounds good to me.